src/components/EventForm.tsx

```diff
@@ -36,92 +53,109 @@ function Index() {
       setVideo(file);
     }
   };
+
   const handleSubmit = async (
     values: MyFormValues,
     actions: FormikHelpers<MyFormValues>
   ) => {
     actions.setSubmitting(false);
     setLoading(true);
-    const myTimestamp = date ? Timestamp?.fromDate(date) : null;
     try {
-      const imageUrl = image && (await uploadImage(image));
-      const data = {
+      const myTimestamp = date ? Timestamp.fromDate(date) : undefined;
+      const imageUrl = image ? await uploadImage(image) : undefined;
+      const data: Partial<Event> = {
         ...values,
         image: imageUrl,
         date: myTimestamp,
       };
-      await addEvent(data);
-      toast.success("A new event has been added successfully ! ", {
-        position: "top-right",
-        autoClose: 5000,
-        hideProgressBar: false,
-        closeOnClick: true,
-        pauseOnHover: true,
-        draggable: true,
-        theme: "colored",
-      });
-    } catch {
-      toast.error(
-        "There was an error adding your event, please try again later",
-        {
-          position: "top-right",
-          autoClose: 5000,
-          hideProgressBar: false,
-          closeOnClick: true,
-          pauseOnHover: true,
-          draggable: true,
-          theme: "colored",
+      // if there is an event it means we are in edit event
+      if (event) {
```

Comment:
Instead of depending on the `event` value here, you could pass a Boolean prop from the edit page for example (e.g. `<EventForm isEditPage />`). This would make the logic more readable and more robust instead of depending the `event` object.